src/PeerPoolMgr.cc

```diff
@@ -225,7 +225,12 @@ PeerPoolMgr::checkpoint(const char *reason)
 void
 PeerPoolMgr::Checkpoint(const Pointer &mgr, const char *reason)
 {
-    CallJobHere1(48, 5, mgr, PeerPoolMgr, checkpoint, reason);
+    if (!mgr.valid())
+        return;
```

Comment:
To simplify condition and add debugging:

```suggestion
    if (!mgr) {
        debugs(48, 5, reason << " but no mgr");
        return;
    }
```

Reply:
Done.